addons/io_scene_gltf2/blender/exp/gltf2_blender_extract.py

```diff
@@ -283,10 +284,90 @@ def extract_primitives(glTF, blender_mesh, library, blender_object, blender_vert
             'material': material_idx,
         })
 
+    if export_settings[gltf2_blender_export_keys.LOOSE_EDGES]:
+        excluded_indices = frozenset([v for p in blender_mesh.polygons for v in p.vertices])
+        line_primitives = extract_line_primitives(blender_mesh, excluded_indices, export_settings)    
+        primitives.extend(line_primitives)
+
+    if export_settings[gltf2_blender_export_keys.LOOSE_VERTICES]:
+        excluded_indices = frozenset([v for e in blender_mesh.edges for v in e.vertices])
+        point_primitives = extract_point_primitives(blender_mesh, excluded_indices, export_settings)    
+        primitives.extend(point_primitives)
+
     print_console('INFO', 'Primitives created: %d' % len(primitives))
 
     return primitives
 
+def extract_line_primitives(blender_mesh, excluded_indices, export_settings):
+    valid_edges = []
+    for blender_edge in blender_mesh.edges:
+        if blender_edge.vertices[0] not in excluded_indices or blender_edge.vertices[1] not in excluded_indices:
+            # edge has at least one edge not in a polygon, add to primitive
+            valid_edges.append(blender_edge)
+    
+    if len(valid_edges) == 0:
+        return []
+    
+    else:
+        vert_count = 0
+        for e in valid_edges:
+            vert_count += len(e.vertices)
+        
+        attributes = {
+            'POSITION': np.empty((vert_count, 3), dtype=np.float32),
+            'NORMAL': np.empty((vert_count, 3), dtype=np.float32)
+        }
```

Comment:
You need to do morph targets (`MORPH_POSITION_%d`) too. Spec requires that all primitives have the same number of targets, so if you don't do it the file will be invalid.

You need WEIGHTS and JOINTS too so skinning will work.

Reply:
Nice to see this can be supported too. 👍🏻